uPortSDK/Classes/Crypto/Crypto.swift

```diff
@@ -0,0 +1,172 @@
+        //Decode base64 public key
+        let boxPubDecoded = boxPub.decodeBase64()
+        
+        //create ephemeral keypair
+        let ephemKeyPair = sodium.box.keyPair()!
+        let ephemPublicKeyString = Data(ephemKeyPair.publicKey).base64EncodedString()
+        
+        //generate random nonce, should be 24 bytes
+        let nonce = sodium.randomBytes.buf(length: sodium.box.NonceBytes)!
+        let nonceString = Data(nonce).base64EncodedString()
+        
+        //pad message
+        let msg = message.padToBlock()
+        //let msgBytes = Bytes(msg.utf8)
+        
+        //seal box
+        let cipherText = sodium.box.seal(message: msg,
+                                         recipientPublicKey: Bytes(boxPubDecoded),
+                                         senderSecretKey: ephemKeyPair.secretKey,
+                                         nonce: nonce)
+        
+        let cipherString = Data(cipherText!).base64EncodedString()
+        
+        //Create encrypted payload object
+        let encPayload = EncryptedMessage(nonce: nonceString,
+                                          ephemPublicKey: ephemPublicKeyString,
+                                          ciphertext: cipherString)
+        
+        return encPayload
+    }
+    
+    /**
+     Decrypts a message with a recipient's secret key.
+     
+     - Parameter encrypted: EncryptedMessage object.
+     - Parameter secretKey: The recipients Secret key as a string
+     
+     - Returns: The decrypted message as a String.
+     */
+    public static func decrypt(encrypted: EncryptedMessage, secretKey: Array<UInt8>) -> String
+    {
+        let sodium = Sodium()
+        let decodedCipherText = Bytes(encrypted.ciphertext.decodeBase64())
+        let decodedEphemPublicKey = Bytes(encrypted.ephemPublicKey.decodeBase64())
+        let decodedNonce = Bytes(encrypted.nonce.decodeBase64())
+        
+        let decrypted = sodium.box.open(authenticatedCipherText: decodedCipherText, senderPublicKey: decodedEphemPublicKey, recipientSecretKey: secretKey, nonce: decodedNonce)!
+        let unpadded = decrypted.unpadFromBlock()
+        return unpadded
+    }
+    
+}
+
+extension String
+{
+    func decodeBase64() -> Data
+    {
+        return Data(base64Encoded: self)!
+    }
+    
+    func padToBlock() -> Array<UInt8>
+    {
+        let bytes: [UInt8] = Array(self.utf8)
+        let paddingSize = Int(ceil(Double(bytes.count) / Crypto.Constants.blockSize ) * Crypto.Constants.blockSize) - bytes.count
+        let padding: Array<UInt8> = Array(repeating: 0, count: paddingSize )
+        return bytes + padding
+    }
+}
+
+extension Array where Element == UInt8
+{
+    func unpadFromBlock() -> String
+    {
+        if let firstZero = self.firstIndex(of: 0)
+        {
+            let unpadded = self[0..<firstZero]
+            return String(bytes: unpadded, encoding: .utf8)!
+        } else
+        {
+            return String(bytes: self, encoding: .utf8)!
```

Comment:
Same: Must be literally 100% sure this can never crash the app via `nil` result.